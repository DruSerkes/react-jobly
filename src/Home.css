.Home {
	width: 100vw;
	height: 90vh;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.Home h1 {
	font-size: 2.5rem;
}

.Home p {
	font-size: 1.3rem;
	font-weight: 300;
}

.Home h2 {
	font-size: 2rem;
	font-weight: 500;
}

.Home button {
	background-color: royalblue;
	margin: 1.8rem;
}

.Home button:hover {
	background-color: rgb(51, 83, 179);
}
