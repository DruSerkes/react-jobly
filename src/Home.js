import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import userContext from './Context';
import './Home.css';

// TODO
// 1. Add styles for Profile components
// 2. Add tests for Home, Login, Profile components
// 3. Step 9 - add apply

const Home = () => {
	const { currentUser } = useContext(userContext);

	return (
		<div className="Home">
			<h1>Jobly</h1>
			<p>All the jobs in one, convenient place.</p>
			{currentUser ? (
				<h2>Welcome Back!</h2>
			) : (
				<Link to="/login">
					<button>Log in</button>
				</Link>
			)}
		</div>
	);
};

export default Home;
