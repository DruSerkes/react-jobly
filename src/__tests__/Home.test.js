import React, { useContext } from 'react';
import userContext from '../Context';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from '../Home';

describe('Home tests', () => {
	it('renders without crashing', () => {
		let currentUser = null;

		render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Home />
				</MemoryRouter>
			</userContext.Provider>
		);
	});

	it('matches snapshot', () => {
		let currentUser = null;

		const { asFragment } = render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Home />
				</MemoryRouter>
			</userContext.Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('welcomes back if currentUser', () => {
		let currentUser = true;
		const { getByText } = render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Home />
				</MemoryRouter>
			</userContext.Provider>
		);
		expect(getByText('Welcome Back!')).toBeInTheDocument();
	});
});
