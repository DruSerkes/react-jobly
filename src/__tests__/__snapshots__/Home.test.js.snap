// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      All the jobs in one, convenient place.
    </p>
    <a
      href="/login"
    >
      <button>
        Log in
      </button>
    </a>
  </div>
</DocumentFragment>
`;
